src/components/com_tjcertificate/administrator/libraries/certificate.php

```diff
@@ -941,4 +944,39 @@ protected function generateUniqueCertId($options)
 
 		return $certificateString;
 	}
+
+	/**
+	 * This function checks the certificate download permission 
+	 *
+	 * @param   STRING  $uniqueCertificateId  certificate Id
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function canDownload($uniqueCertificateId)
```

Comment:
You don't need to pass any parameter. use $this->uniqueCertificateId inside function

Reply:
Done, changed to static function